### PR TITLE
chore(POC) using native kubectl kustomize during deployment

### PR DESCRIPTION
**Pull request by Voronenko**
### What does this PR do?

introduces customization of parameters during deployment

### CODE REVIEW

File: deployment/kubernetes/k8s/base/agent-config-map.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+kind: ConfigMap
+metadata:
+  name: sts-agent-config
+  namespace: default
+data:
+  STS_API_KEY: "API_KEY"
+  STS_STS_URL: "<STACKSTATE_BACKEND_URL>/stsAgent"
+  STS_PROCESS_AGENT_URL: "<STACKSTATE_BACKEND_URL>/stsAgent"
+  STS_APM_URL: "<STACKSTATE_BACKEND_URL>/stsAgent"
+  KUBERNETES: "true"
```
Comment (TizianoPerrucci) [inline deployment/kubernetes/k8s/base/agent-config-map.yaml]: move this out of config map

File: deployment/kubernetes/k8s/base/agent-config-map.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  STS_PROCESS_AGENT_ENABLED: "true"
+  STS_APM_ENABLED: "true"
+  STS_NETWORK_TRACING_ENABLED: "true"
+  STS_SKIP_SSL_VALIDATION: "false"
+  HOST_PROC: "/host/proc"
```
Comment (TizianoPerrucci) [inline deployment/kubernetes/k8s/base/agent-config-map.yaml]: move `HOST_*` out of config map

File: deployment/kubernetes/k8s/base/cluster-auth-token.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: stackstate-auth-token
+type: Opaque
+data:
+  # You can generate with: `echo -n '<ThirtyX2XcharactersXlongXtoken>' | base64`
+  token: PFRoaXJ0eVgyWGNoYXJhY3RlcnNYbG9uZ1h0b2tlbj4=
```
Comment (TizianoPerrucci) [inline deployment/kubernetes/k8s/base/cluster-auth-token.yaml]: could we override this token when applying manifests with kustomize ?

File: deployment/kubernetes/k8s/overlays/example/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+configMapGenerator:
+- name: sts-agent-config
+  behavior: merge
+  literals:
+     - STS_API_KEY=SOMEOTHER
+- name: sts-clusteragent-config
+  behavior: merge
+  literals:
+     - STS_API_KEY=SOMEOTHER2
```
Comment (TizianoPerrucci) [inline deployment/kubernetes/k8s/overlays/example/kustomization.yaml]: add `STS_STS_URL` and `STS_CLUSTER_NAME` to overrides

File: deployment/kubernetes/k8s/overlays/example/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+kind: Kustomization
+
+bases:
+- ../../base
+
+configMapGenerator:
+- name: sts-agent-config
+  behavior: merge
+  literals:
+     - STS_API_KEY=SOMEOTHER
```
Comment (TizianoPerrucci) [inline deployment/kubernetes/k8s/overlays/example/kustomization.yaml]: add `STS_STS_URL` `STS_PROCESS_AGENT_URL` `STS_APM_URL` to the overrides.
can we avoid specifying all of them and just use one variable (eg. `STS_URL`) ?

File: deployment/kubernetes/README.md [hunk 6]
```diff
@@ -23,6 +23,74 @@ To gather your kube-state metrics:
...
+Create dedicated configuration directory which contains necessary overrides over default variables, named `kustomization.yaml`
+As a minimum, overrides include
+
+* `<STS_API_KEY>` with your StackState backend URL
+* `<STS_STS_URL>` with your StackState backend URL (including suffix /stsAgent)
+* `<STS_CLUSTER_NAME>` with your Cluster name
+
+
+
+```yaml
```
Comment (TizianoPerrucci) [inline deployment/kubernetes/README.md]: remove this yaml snippet, just link to the `example/kustomization.yaml`

File: deployment/kubernetes/k8s/overlays/example/kustomization.yaml (hunk 5 above)
Comment (TizianoPerrucci) [reply deployment/kubernetes/k8s/overlays/example/kustomization.yaml]: @Voronenko `STS_PROCESS_AGENT_URL` `STS_APM_URL` are still missing in the agent-config overrides.
Did try to find a way to set the backend url only once?

File: deployment/kubernetes/k8s/base/cluster-auth-token.yaml (hunk 3 above)
Comment (Voronenko) [reply deployment/kubernetes/k8s/base/cluster-auth-token.yaml]: Yes, I've introduced via SecretMapGenerator, but I think it also can be replaced as file fragment

File: deployment/kubernetes/k8s/overlays/example/kustomization.yaml (hunk 5 above)
Comment (Voronenko) [reply deployment/kubernetes/k8s/overlays/example/kustomization.yaml]: Reliable - no. There is concept of "variables"  (kustomization_explained.yaml in kustomization folder) - but does not work smoothly.

Ideally, if url issue could be resolved inside go agent itself. (Like prefer  STS_PROCESS_AGENT_URL , but use STS_STS_URL if not provided(

**Pull request merged**